### PR TITLE
Add support for Python 3.11.

### PR DESCRIPTION
**Pull request by icemac**
- Lint the code.
- Improve error handling in test.
- ~~Tests currently fail on Python 2.7 -- I have no clue how this ever can have worked.~~ Drop support for Python < 3.7.

### CODE REVIEW

**Comment by icemac** [comment]
This PR is here to update #29 with it, to see what will happen when activating untrusted python for 3.11.

**Comment by mgedmin** [comment]
> Tests currently fail on Python 2.7 -- I have no clue how this ever can have worked.

I'm also seeing failures from `tox -e py27` on the current git master.

How did GitHub Actions [succeed on 2.7 four days ago](https://github.com/zopefoundation/zope.pagetemplate/actions/runs/3558096822/jobs/5976528388)?

**Comment by icemac** [comment]
> How did GitHub Actions [succeed on 2.7 four days ago](https://github.com/zopefoundation/zope.pagetemplate/actions/runs/3558096822/jobs/5976528388)?

This is a mystery for me, too. At #29 I fixed the issue by rewriting the test. As it is not clear to me how this ever can have worked in the first place: names starting with `_` are forbidden in `RestrictedPython` forever.

File: src/zope/pagetemplate/pagetemplate.py [hunk 1]
```diff
@@ -172,7 +170,7 @@ def write(self, text):
         # We accept both, since the text can either come from a file (and the
         # parser will take care of the encoding) or from a TTW template, in
         # which case we already have unicode.
-        assert isinstance(text, (six.string_types, six.binary_type))
+        assert isinstance(text, ((str,), bytes))
```
Comment (gforcada) [inline src/zope/pagetemplate/pagetemplate.py:173]: Should that ben `isinstance(text, (str, bytes))`

File: src/zope/pagetemplate/engine.py [hunk 2]
```diff
@@ -19,15 +19,19 @@
...
-from zope.i18n import translate
+from zope.traversing.interfaces import IPathAdapter
+from zope.traversing.interfaces import ITraversable
+from zope.traversing.interfaces import TraversalError
+
+from zope import component
+
 
 try:  # pragma: no cover
     # Until https://github.com/zopefoundation/zope.untrustedpython/issues/2
```
Comment (gforcada) [inline src/zope/pagetemplate/engine.py:37]: This issue is closed already, not sure if that means that this `try/except` can be removed though 🤔

File: src/zope/pagetemplate/pagetemplate.py [hunk 3]
```diff
@@ -209,9 +210,8 @@ def bs(s):
                            (self._error_start, "%s: %s" %
                             sys.exc_info()[:2])) + self._text)
 
-        return bs('%s\n %s\n-->\n' % (self._error_start,
-                                      '\n'.join(self._v_errors))) + \
-            self._text
+        return bs('{}\n {}\n-->\n'.format(
+            self._error_start, '\n'.join(self._v_errors))) + self._text
```
Comment (gforcada) [inline src/zope/pagetemplate/pagetemplate.py:214]:
```suggestion
        return bs('{}\n {}\n-->\n{}'.format(
            self._error_start, '\n'.join(self._v_errors), self._text))
```

Would that be the same, right?

**Review by gforcada — APPROVED** [review]

File: src/zope/pagetemplate/pagetemplate.py (hunk 3 above)
Comment (icemac) [reply src/zope/pagetemplate/pagetemplate.py:214]: Right.

File: src/zope/pagetemplate/pagetemplate.py (hunk 1 above)
Comment (icemac) [reply src/zope/pagetemplate/pagetemplate.py:173]: Already fixed it locally but forgot to commit it.

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**